File: e-commerce_backend/src/main/java/ecart/ecommerce/service/ProductService.java

package ecart.ecommerce.service;

import ecart.ecommerce.entity.Product;
import ecart.ecommerce.enums.ProductStatus;

import java.util.List;

public interface ProductService {
    Product addProduct(Product product);
    Product getProductById(String id);
    List<Product> getAllProducts();
    List<Product> getProductsBySeller(String sellerId);
    List<Product> getProductsByCategory(String category);
    List<Product> getProductsByStatus(ProductStatus status);
    List<Product> searchProductsByName(String name);
    Product updateProduct(String id,Product product);
    Product updateProductStatus(String id,ProductStatus status);
    void deleteProduct(String id);
    
}

File: e-commerce_Frontend/src/service/productService.js
import api from "./api";

export const addProduct = async (product) => {
    const response = await api.post(
        "/products",
        product
    );

    return response.data;
};

export const getAllProducts = async () => {
    const response = await api.get(
        "/products"
    );

    return response.data;
};

export const getProductById = async (id) => {
    const response = await api.get(
        `/products/${id}`
    );

    return response.data;
};

export const getProductsBySeller = async (sellerId) => {
    const response = await api.get(
        `/products/seller/${sellerId}`
    );

    return response.data;
};

export const updateProduct = async (
    id,
    product
) => {
    const response = await api.put(
        `/products/${id}`,
        product
    );

    return response.data;
};

export const deleteProduct = async (id) => {
    const response = await api.delete(
        `/products/${id}`
    );

    return response.data;
};